Source/Readers/ImageReader/ImageTransformers.cpp

```diff
@@ -368,54 +370,56 @@ StreamInformation ScaleTransformer::Transform(const StreamInformation& inputStre
     return m_outputStream;
 }
 
-void ScaleTransformer::Apply(uint8_t, cv::Mat &mat, int /* indexInBatch */)
+void ScaleTransformer::Apply(uint8_t, cv::Mat& mat, int /* indexInBatch */)
 {
     if (m_scaleMode == ScaleMode::Fill)
     { // warp the image to the given target size
-        cv::resize(mat, mat, cv::Size((int)m_imgWidth, (int)m_imgHeight), 0, 0, m_interp);
+        cv::resize(mat, mat, cv::Size((int) m_imgWidth, (int) m_imgHeight), 0, 0, m_interp);
     }
     else
     {
         int height = mat.rows;
         int width = mat.cols;
 
         // which dimension is our scaled one?
+        const double widthFactor = (double) m_imgWidth / (double) width;
+        const double heightFactor = (double) m_imgHeight / (double) height;
         bool scaleW;
         if (m_scaleMode == ScaleMode::Crop)
-            scaleW = width < height; // in "crop" mode we resize the smaller side
+            scaleW = widthFactor > heightFactor; // in "crop" mode we resize along the larger ratio
         else
-            scaleW = width > height; // else we resize the larger side
+            scaleW = widthFactor < heightFactor; // else we resize along the smaller ratio
 
         size_t targetW, targetH;
         if (scaleW)
         {
-            targetW = (size_t)m_imgWidth;
-            targetH = (size_t)round(height * m_imgWidth / (double)width);
+            targetW = (size_t) m_imgWidth;
```

Comment:
Can you please remove changes like this one (and several others below) where you are introducing space in the casting operation. The original code is fine as far as coding style goes, and it is sort of obscuring the real changes in this PR.